site/_posts/2023-03-09-react-example-application.md

```diff
@@ -0,0 +1,309 @@
+With `RequireAuth` you can wrap a component and require the user to be authenticated before the component is rendered. The `withRole` option also enables you to restrict the user to a specific role.
+
+`RequireAuth` is used in the `LoggedInMenu` component, which is used to display the user information, profile, and logout button if the user is authenticated.
+
+```tsx
+{% remote_include 'https://raw.githubusercontent.com/sonderformat-llc/fusionauth-example-react-fusiondesk/main/client/src/components/LoggedInMenu.tsx' %}
+```
+
+With the `FusionAuthConfig` you can configure the FusionAuth instance that is used by the React SDK. If you are using a different OAuth 2.0 backend, you will most likely have to configure the route props.
+
+We are using the React Router v6 in the example application. To protect routes, we are using a custom `ProtectedRoutes` component. This component uses the `useFusionAuth` hook to check if the user is authenticated. If the user is authenticated, the component renders the `Outlet` component. If the user is not authenticated, the component redirects the user to the `/` route.
+
+```tsx
+{% remote_include 'https://raw.githubusercontent.com/sonderformat-llc/fusionauth-example-react-fusiondesk/main/client/src/components/ProtectedRoutes.tsx' %}
+```
+
+{% include _callout-note.liquid
+content=
+"For more information about the React SDK, see the [FusionAuth React SDK](/docs/v1/tech/client-libraries/react-sdk) documentation."
+%}
+
+### Login Page
+
+{% include _image.liquid src="/assets/img/blogs/react-example-application/fusiondesk-login.png" alt="Login / Register Prompt if the user is not authenticated" class="img-fluid" figure=false %}
+
+The login page allows the user to login or register using the FusionAuth login / register flow. The login page is implemented in the `LoginPage.tsx` file and displays the `login` and `register` buttons.
+To use the button styling provided by DaisyUI instead of the one provided by FusionAuth, we are not using the pre-built buttons from the React SDK. Instead, we are using the `useFusionAuth` hook to get the `login` and `register` methods from the `FusionAuthContext`.
+
+```tsx
+{% remote_include 'https://raw.githubusercontent.com/sonderformat-llc/fusionauth-example-react-fusiondesk/main/client/src/pages/LoginPage.tsx' %}
+```
+
+### Tickets Page
+
+{% include _image.liquid src="/assets/img/blogs/react-example-application/fusiondesk-tickets.png" alt="Overview of the tickets when the user is authenticated" class="img-fluid" figure=false %}
+
+The tickets page displays the tickets of the user. The tickets are retrieved from the backend using the `/api/tickets` endpoint. The returned tickets depend on the role of the user. If the user is only a `customer`, only the tickets of the user are returned.
+
+We also embed the user information in the backend response. This allows us to display the creator information of the ticket (full name, picture).
+
+Backend `GET /` endpoint:
+
+```typescript
```

Comment:
Maybe we could have endpoints in separated files so we can use `remote_include` here as well?